airflow/www/views.py

```diff
@@ -3012,6 +3015,55 @@ def action_muldelete(self, items):
         self.update_redirect()
         return redirect(self.get_redirect())
 
+    @action(
+        'mulduplicate',
+        'Duplicate',
+        'Are you sure you want to duplicate the selected connections?',
+        single=False,
+    )
+    @provide_session
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_CREATE, permissions.RESOURCE_CONNECTION),
+        ]
+    )
+    def action_mulduplicate(self, connections, session=None):
+        """Duplicate Multiple connections"""
+        for selected_conn in connections:
+            new_conn_id = selected_conn.conn_id
+            match = re.search(r"_copy(\d+)$", selected_conn.conn_id)
+            if match:
+                conn_id_prefix = selected_conn.conn_id[: match.start()]
+                new_conn_id = f"{conn_id_prefix}_copy{int(match.group(1)) + 1}"
+            else:
+                new_conn_id += '_copy1'
+
+            dup_conn = Connection(
+                new_conn_id,
+                selected_conn.conn_type,
+                selected_conn.description,
+                selected_conn.host,
+                selected_conn.login,
+                selected_conn.password,
+                selected_conn.schema,
+                selected_conn.port,
+                selected_conn.extra,
+            )
+
+            try:
+                session.add(dup_conn)
+                session.commit()
+                flash(f"Connection {new_conn_id} added successfully.", "success")
+            except IntegrityError:
+                flash(
+                    f"Connection {new_conn_id} can't be added. Integrity error, probably unique constraint.",
+                    "warning",
+                )
+                session.rollback()
```

Comment:
Some thoughts on the current approach inserting the duplicated connections:

* This commits each duplicated connection separately, so any failures would not affect other duplications. I’m not sure I’d want this as a user (probably subjective).
* The error message is not very helpful either. We could make a bit more effort by querying whether there’s an existing connection to the ID, but that could be subject to race conditions as well? (I think some database-level configuration can avoid this?)
* Combining the previous two points, it may be possible to first connect all the new IDs, query whether any of them exist, and fail all of them cause conflicts or fail to insert.
* If there’s more than one failures, the user may get a bunch of flashes which may be difficult to read? But maybe that’s not really an issue in practice.

Reply:
@uranusjr,
I think the current approach is better, as all the connections are independent to each other so they will get independent status. 
Also, from an error message point of view, We are only catching integrity error, so we are providing the same error.
A similar approach is already in place in add() for the connection.